packages/office-ui-fabric-react/src/components/DetailsList/DetailsRowFields.tsx

```diff
@@ -43,25 +44,35 @@ export class DetailsRowFields extends React.PureComponent<IDetailsRowFieldsProps
                 cellStyleProps.cellRightPadding +
                 (column.isPadded ? cellStyleProps.cellExtraRightPadding : 0);
 
-          const { onRender = onRenderItemColumn } = column;
+          const { onRender = onRenderItemColumn, getValueKey = getCellValueKey } = column;
           const cellContentsRender =
             cellsByColumn && column.key in cellsByColumn
               ? cellsByColumn[column.key]
               : onRender
               ? onRender(item, itemIndex, column)
               : getCellText(item, column);
 
+          // generate a key that auto-dirties when content changes, to force the container to re-render, to trigger animation
+          let key: number | string;
+          try {
```

Comment:
why is a try catch needed here?

Reply:
original thinking must be that if user passed in callback caused exception should not impact the rendering of list?

Reply:
perhaps we should trust this just as trust the rest of callback?!